test/data/lightning/test_datamodule.py

```diff
@@ -302,6 +302,7 @@ def test_lightning_hetero_node_data(get_dataset):
     assert torch.all(data['paper'].x > original_x)  # Ensure shared data.
     assert trainer.validate_loop._data_source.is_defined()
     assert trainer.test_loop._data_source.is_defined()
+    torch.distributed.destroy_process_group()
```

Comment:
Thanks for fixing this! In order to make sure it runs even when the test case fails before the line, could we create a pytest fixture that destroys the pg on teardown?

Reply:
Done. Thanks for the advice!

Reply:
Please delay approval. I see another problem with this test and I need more time to fix it.